number_theory/modular.h

```diff
@@ -132,6 +134,19 @@ class Modular {
   }
 };
 
+// Returns the modular inverse of |number| in modulo |mod| if exists.
+// Otherwise, throws an invalid_argument exception.
+template <typename T,
+          std::enable_if_t<std::numeric_limits<T>::is_integer, bool> = true>
+T inverse(T number, T mod) {
+  T num = modular_internal::normalize(std::move(number), mod);
+  auto [x, y] = exgcd(num, mod);
+  if (x * num + y * mod != 1) {
+    throw std::invalid_argument("The inverse does not exist.");
```

Comment:
What about printing the values of `number` and `mod` with the error message?

Reply:
> What about printing the values of `number` and `mod` with the error message?

It will be nice.

Reply:
I think it will make the code significantly messier unless we implement a wrapper for it. People suggest interesting approaches here:
https://stackoverflow.com/questions/12261915/how-to-throw-stdexceptions-with-variable-messages

Reply:
We can use `std::format`?